src/Form/Type/PreAddAssetType.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Form\Type;
+
+use App\Entity\AssetType;
+use App\Entity\Organization;
+use App\Repository\AssetTypeRepository;
+use LogicException;
+use Symfony\Bridge\Doctrine\Form\Type\EntityType;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\SubmitType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Security\Core\Security;
+
+class PreAddAssetType extends AbstractType
+{
+    protected AssetTypeRepository $assetTypeRepository;
+    protected Organization $organization;
+
+    public function __construct(Security $security, AssetTypeRepository $assetTypeRepository)
+    {
+        $this->assetTypeRepository = $assetTypeRepository;
+
+        /** @var Organization $currentUser */
+        $currentUser = $security->getUser();
+        if (!$currentUser instanceof Organization) {
+            throw new LogicException('Current user must be an organization');
+        }
+
+        $this->organization = $currentUser;
```

Comment:
this feels strange that this form is not stateless: we may add `parent_organization` in the option resolver as a required field (we also can add a callback validator on it to ensure it is a parent organizatoin)